xds/src/main/java/io/grpc/xds/EnvoyProtoData.java

```diff
@@ -339,4 +341,239 @@ public String toString() {
           .toString();
     }
   }
+
+  /** See corresponding Envoy proto message {@link io.envoyproxy.envoy.api.v2.route.Route}. */
+  static final class Route {
+    private final RouteMatch routeMatch;
+    @Nullable
+    private final RouteAction routeAction;
+
+    @VisibleForTesting
+    Route(RouteMatch routeMatch, @Nullable RouteAction routeAction) {
+      this.routeMatch = routeMatch;
+      this.routeAction = routeAction;
+    }
+
+    RouteMatch getRouteMatch() {
+      return routeMatch;
+    }
+
+    Optional<RouteAction> getRouteAction() {
```

Comment:
I would just prefer using nullable return value and all the other places used that (it doesn't give us much benefit for using `Optional`). But I am fine with this.